src/ert/scheduler/job.py

```diff
@@ -78,6 +85,8 @@ async def __call__(
             await self.started.wait()
 
             await self._send(State.RUNNING)
+            if self.real.max_runtime is not None and self.real.max_runtime > 0:
+                cancel_task = asyncio.create_task(self._max_runtime_task())
```

Comment:
This is a sidenote and I think `asyncio.wait_for` is better solution overall, but I discovered that you can do `asyncio.current_task()` to get the current `asyncio.Task`, which you can pass to a cancel task.